src/main/java/com/rackspacecloud/metrics/queryservice/services/QueryServiceImpl.java

```diff
@@ -169,6 +177,21 @@ private QueryResult getDataPointsForGivenQuery(String tenantId, String queryStri
         return qr;
     }
 
+    private String getCleanedMeasurementName(String queryString, String measurement) {
+        // Measurement name coming from Grafana is wrapped up in double quotes. So, we need to remove those
+        // double quote characters to get the actual measurement name string
+        // For example: Measurement name foo_bar from Grafana will show up as "foo_bar"
+        if(measurement.startsWith("\"") && measurement.endsWith("\"")) {
+            measurement = measurement.substring(1, measurement.lastIndexOf("\""));
```

Comment:
Instead of using lastindexof, consider using measurement.length()-2 to avoid a search

Reply:
👍

Reply:
it should be "measurement.length()-1"